File: practice/algorithms/implementation/repeated_string/RepeatedString.scala
import scala.io.Source

object RepeatedString extends App {
  val lines = Source.stdin.getLines().toList
  val string = lines.head
  val letters = lines(1).toLong
  val as = string.count(_ == 'a')
  val times = letters / string.length
  val rest = letters % string.length

  val totalAs = times * as + string.substring(0, rest.toInt).count(_ == 'a')
  println(totalAs)
}

File: practice/algorithms/implementation/repeated_string/RepeatedString.java
import java.util.*;

public class RepeatedString {

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        String s = stdin.nextLine();
        long n = Long.parseLong(stdin.nextLine());
        long as = s.chars().filter(c -> c == 'a').count();
        long times = n / s.length();
        long rest = n % s.length();

        long totalAs = times * as
                + s.substring(0, (int) rest).chars().filter(c -> c == 'a').count();
        System.out.println(totalAs);
        stdin.close();
    }
}